### PR TITLE
add key interface

### PR DESCRIPTION
**Pull request by MaggieNgWu**

### CODE REVIEW

File: libdevcrypto/sdf/SDFCryptoProvider.h [hunk 1]
```diff
@@ -19,12 +19,35 @@
...
  */
 
+#include <libdevcrypto/Common.h>
 #include <string>
-
 namespace dev
 {
 namespace crypto
 {
+enum Algorithm : uint32_t
```
Comment (cyjseagull) [inline libdevcrypto/sdf/SDFCryptoProvider.h:28]: recommend `Algorithm` to `AlgorithmType`

File: libdevcrypto/sdf/SDFCryptoProvider.h [hunk 2]
```diff
@@ -19,12 +19,35 @@
...
+    Key(unsigned int keyIndex, std::string password)
+    {
+        m_key_index = keyIndex;
+        m_key_password = (unsigned char*)password.c_str();
+    };
+    unsigned int Identifier() { return m_key_index; };
+    unsigned char* Password() { return m_key_password; };
+
+private:
+    unsigned int m_key_index;
```
Comment (cyjseagull) [inline libdevcrypto/sdf/SDFCryptoProvider.h:47]: `m_key_index`->`m_keyIndex`
`m_key_password`->`m_keyPassword`

File: libdevcrypto/sdf/SDFCryptoProvider.h [hunk 3]
```diff
@@ -19,12 +19,35 @@
...
+    unsigned char* m_key_password;
```
Comment (cyjseagull) [inline libdevcrypto/sdf/SDFCryptoProvider.h:48]: `unsigned char*` to `string`

File: libdevcrypto/sdf/SDFCryptoProvider.h [hunk 4]
```diff
@@ -58,37 +79,37 @@ class SDFCryptoProvider
...
      * Return error code
      */
-    unsigned int KeyGen(std::string algorithm, Key* key);
+    unsigned int KeyGen(Algorithm algorithm, Key* key);
 
     /**
      * Sign
      */
-    unsigned int Sign(Key const* key, std::string algorithm, unsigned char const* digest,
+    unsigned int Sign(Key const* key, Algorithm algorithm, unsigned char const* digest,
```
Comment (cyjseagull) [inline libdevcrypto/sdf/SDFCryptoProvider.h:87]: Recommend `Key const*` to `shared_ptr<Key>` or `Key const&`

**Review by cyjseagull — APPROVED** [review]

**Review by bxq2011hust — APPROVED** [review]

**Pull request merged**